src/lockscreen.js

```diff
@@ -116,21 +162,14 @@ var Lockscreen = GObject.registerClass({
         if (!this._overlay)
             return;
 
-        const style = this._overlay.get_style_context();
         if (this._locked) {
-            style.add_class('locked');
-            this.add_overlay(this._overlay);
+            this._style.add_class('locked');
+            this._playbin.stop();
         } else {
-            style.remove_class('locked');
-
-            // hide widget after transition completes, so that it doesn't
-            // continue to intercept clicks
-            GLib.timeout_add(GLib.PRIORITY_DEFAULT, FADE_OUT_TIME_MS, () => {
-                this.remove(this._overlay);
-                return GLib.SOURCE_REMOVE;
-            });
+            this._playbin.play();
```

Comment:
Keep in mind that, the animation should only be displayed under the specific condition where 1) the lock-screen was being displayed as locked, 2) the user by clicking or maybe the clubhouse by interacting with the GSS, has update the lock GSS state to be unlocked.

So we shouldn't be playing or stopping the animation here.